frontends/web/src/routes/account/send/send.tsx

```diff
@@ -249,18 +245,23 @@ class Send extends Component<Props, State> {
     }
   };
 
-  private txInput = () => ({
-    address: this.state.recipientAddress,
-    amount: this.state.amount,
-    feeTarget: this.state.feeTarget || '',
-    customFee: this.state.customFee,
-    sendAll: this.state.sendAll ? 'yes' : 'no',
-    selectedUTXOs: Object.keys(this.selectedUTXOs),
-  });
-
-  private sendDisabled = () => {
-    const txInput = this.txInput();
-    return !txInput.address || this.state.feeTarget === undefined || (txInput.sendAll === 'no' && !txInput.amount) || (this.state.feeTarget === 'custom' && !this.state.customFee);
+  private getValidTxInputData = (): Required<accountApi.TTxInput> | false => {
+    if (
+      !this.state.recipientAddress
+      || this.state.feeTarget === undefined
+      || (!this.state.sendAll && !this.state.amount)
+      || (this.state.feeTarget === 'custom' && !this.state.customFee)
+    ) {
+      return false;
+    }
+    return {
+      address: this.state.recipientAddress,
+      amount: this.state.amount,
+      feeTarget: this.state.feeTarget,
+      customFee: this.state.customFee,
+      sendAll: (this.state.sendAll ? 'yes' : 'no'),
+      selectedUTXOs: Object.keys(this.selectedUTXOs),
+    };
```

Comment:
Maybe I'm missing smt, but why are we returning this object exactly? Why not just return `true`? 🤔  And if so, change fn name to like `isTxInputDataValid` or something.

Reply:
if we return true we run into this problem again https://github.com/digitalbitbox/bitbox-wallet-app/pull/2678#discussion_r1587746718

Reason: typescript doesn't pass type information it knows from inside a fn to the scope where it was called. So a condition like this `if (fn() {` would again need to have another check so it can ensure that feetarget is not undefined.

does that make sense?

Reply:
Ah, okay. Thanks!